Userland/Applications/ThemeEditor/MainWidget.cpp

```diff
@@ -0,0 +1,421 @@
+            if (result.is_error())
+                return;
+            save_to_file(result.value());
+        }
+    });
+    TRY(file_menu->try_add_action(*m_save_action));
+
+    TRY(file_menu->try_add_action(GUI::CommonActions::make_save_as_action([&](auto&) {
+        auto result = FileSystemAccessClient::Client::the().try_save_file(&window, "Theme", "ini", Core::OpenMode::ReadWrite | Core::OpenMode::Truncate);
+        if (result.is_error())
+            return;
+        save_to_file(result.value());
+    })));
+
+    TRY(file_menu->try_add_separator());
+    TRY(file_menu->try_add_action(GUI::CommonActions::make_quit_action([&](auto&) { GUI::Application::the()->quit(); })));
+
+    auto accessibility_menu = TRY(window.try_add_menu("&Accessibility"));
+
+    auto default_accessibility_action = GUI::Action::create_checkable("Default - non-impaired", { Mod_AltGr, Key_1 }, [&](auto&) {
+        m_preview_widget->set_color_filter(nullptr);
+    });
+    default_accessibility_action->set_checked(true);
+
+    auto pratanopia_accessibility_action = GUI::Action::create_checkable("Protanopia", { Mod_AltGr, Key_2 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_protanopia());
+    });
+
+    auto pratanomaly_accessibility_action = GUI::Action::create_checkable("Protanomaly", { Mod_AltGr, Key_3 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_protanomaly());
+    });
+
+    auto tritanopia_accessibility_action = GUI::Action::create_checkable("Tritanopia", { Mod_AltGr, Key_4 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_tritanopia());
+    });
+
+    auto tritanomaly_accessibility_action = GUI::Action::create_checkable("Tritanomaly", { Mod_AltGr, Key_5 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_tritanomaly());
+    });
+
+    auto deuteranopia_accessibility_action = GUI::Action::create_checkable("Deuteranopia", { Mod_AltGr, Key_6 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_deuteranopia());
+    });
+
+    auto deuteranomaly_accessibility_action = GUI::Action::create_checkable("Deuteranomaly", { Mod_AltGr, Key_7 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_deuteranomaly());
+    });
+
+    auto achromatopsia_accessibility_action = GUI::Action::create_checkable("Achromatopsia", { Mod_AltGr, Key_8 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_achromatopsia());
+    });
+
+    auto achromatomaly_accessibility_action = GUI::Action::create_checkable("Achromatomaly", { Mod_AltGr, Key_9 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_achromatomaly());
+    });
+
+    auto preview_type_action_group = make<GUI::ActionGroup>();
```

Comment:
This probably shouldn’t be a local object. The actions are not exclusive.
![](https://user-images.githubusercontent.com/16520278/167257813-c9a17165-a2d9-4bd0-b314-ffa217e15cc8.png)

Reply:
Thanks! I assumed the ActionGroup would be kept alive by something but it's not.